lec12-numerical-models.Rmd

```diff
@@ -0,0 +1,902 @@
+        behaviour of the system could be very different for starting conditions that were very similar. 
+        I wrote a quick simulation in R which you can play around with, it's at the bottom of these notes.
+        - example: Lotka-volterra predator-prey model. 
+        This model predicts never-ending oscillations in the populations of predator and prey species.
+        A simulation for this is also at the bottom of these notes.
+            - What else is a reasonable thing that could happen to a population of predator and prey? 
+            
+          <!--Possible answers: 
+          - predators eat all the prey and the prey dies, then the predator dies. 
+          - predators and prey reach an unchanging steady-state -->
+
+        *Aside*: I do theoretical biophysics, which is quite similar to theoretical ecology in my case, 
+        so all of what I'll be telling you about is techniques and concepts that I have found useful 
+        or use very frequently and that I think are an important part of the theorist / data scientist toolkit. 
+
+3. In the next two lectures we will talk about population models and about using R to analyze models. 
+We will first do this qualitatively (no math), then quantitatively. We will talk about fitting models to data,
+and by the end I hope that you will be able to incorporate modelling into your project, 
+whether a model you design yourself or an existing ecological model. 
+
+## Setting up a population model
+
+Let's look at some examples of one-dimensional models and talk about what we can learn from these models
+qualitatively, without doing any calculations.
+
+**Dimension**: the number of variables in your model. (Not the number of parameters.  
+Variables are what track the species you're interested in, 
+and parameters are numbers that specify the details of the model.)
+
+Let's write down a model for the growth of a species. We want to predict how many organisms are present 
+in the population at time $t$, and to do that we need to write down an equation that describes how the
+population changes with time. We will start by writing a *recursion relation*, which tells us how 
+to update the population size at time $t+ \Delta t$ from the population size at time $t$:
+
+$$N_{t+\Delta t} = N_t + \Delta N$$
+
+$\Delta N$, the change in the population size, will depend on the specifics of our model. Let's rearrange this
+expression into the definition of a derivative by moving $N_t$ over and dividing both sides by $\Delta t$,
+then taking the limit of $\Delta t$ going to $0$:
+
+$$\frac{N_{t+\Delta t} - N_t}{\Delta t} = \frac{\Delta N}{\Delta t}$$
+$$lim_{\Delta t \to 0}\frac{N_{t+\Delta t} - N_t}{\Delta t} = lim_{\Delta t \to 0}\frac{\Delta N}{\Delta t} = \frac{dN}{dt}$$ 
+
+If you've taken calculus, you will recognize this as the first derivative of $N$ with respect to $t$. 
+If your calculus is foggy in your memory, when you see this notation think "change in $N$ per change in $t$.
```

Comment:
Do you think it's worth relating this back to what we taught them about the interpretation of beta coefficients from linear models? (i.e. the change in `y` for every one unit change in `x`.

Reply:
Good idea!